VerilogCodingStyle.md

```diff
@@ -2291,6 +2292,109 @@ always_comb begin
 end
 ```
 
+### Functions and Tasks
+
+The following section applies to synthesizable RTL only.
+
+***In synthesizable RTL use of functions is allowed, provided they are declared
+automatic. Tasks should not be used.***
+
+Functions must be declared outside of a class in either a package or inside a
+module. A package is appropriate where the function relates to other definitions
+in the package and could be useful to multiple modules (even if its currently
+only used by one). A module is appropriate where the function specifically
+relates to the internals of that module.
+
+Functions should aim to conceptually represent a reusable block of combinational
+logic.
+
+Storage types must be explicitly declared for all arguments and the function
+return value.
+
+Do not use `input`, `output`, `inout`, or `ref` on function arguments.
+
+Do not assign to function arguments in the function body.
```

Comment:
This will be a function of whether we allow `output` above...

Reply:
agree: redundant with above guidance, can be removed.